src/lucky/routable.cr

```diff
@@ -323,6 +322,26 @@ module Lucky::Routable
       Lucky::RouteHelper.new {{ method }}, path
     end
 
+    def self.route(*_args, **_named_args) : Lucky::RouteHelper
+      {% requireds = path_params.map { |param| "#{param.gsub(/:/, "").id} (Path Variable)" } %}
+      {% params_without_defaults.each { |param| requireds << "#{param.var} (Param)" } %}
+      {% optionals = optional_path_params.map { |param| "#{param.gsub(/:/, "").id} (Path Variable)" } %}
+      {% params_with_defaults.each { |param| optionals << "#{param.var} (Param)" } %}
+      \{% raise <<-ERROR
+        Invalid usage of route {{ @type }}
+
+        {% if !requireds.empty? %}
+        Required arguments:
+        {% for req in requireds %}\n- {{ req.id }}{% end %}
+        {% end %}{% if !optionals.empty? %}
+        Optional arguments:
+        {% for opts in optionals %}\n- {{ opts.id }}{% end %}
+        {% end %}
```

Comment:
I looked into doing something like this but I'd be afraid that if I put that in the error message that it would be an indication of the correct way to call it but the `[ {{ optional }} ]` part is not right.